tests/test_phone_numbers_validator.py

```diff
@@ -0,0 +1,157 @@
+
+
+@pytest.mark.parametrize('invalid_number', ['', '123', 12, object(), '55 121'])
+def test_invalid_phone_number(invalid_number: Any) -> None:
+    # Use a TypeAdapter to test the validation logic for None otherwise
+    # optional fields will not attempt to validate
+    with pytest.raises(ValidationError, match='value is not a valid phone number'):
+        number_adapter.validate_python(invalid_number)
+
+
+def test_formats_phone_number() -> None:
+    result = Numbers(phone_number='+1 901 555 1212 ext 12533', uk_number='+44 20 7946 0958')
+    assert result.phone_number == 'tel:+1-901-555-1212;ext=12533'
+    assert result.uk_number == '+442079460958'
+
+
+def test_default_region() -> None:
+    result = Numbers(na_number='901 555 1212')
+    assert result.na_number == 'tel:+1-901-555-1212'
+    with pytest.raises(ValidationError, match='value is not a valid phone number'):
+        Numbers(phone_number='901 555 1212')
+
+
+def test_supported_regions() -> None:
+    assert Numbers(na_number='+1 901 555 1212')
+    assert Numbers(uk_number='+44 20 7946 0958')
+    with pytest.raises(ValidationError, match='value is not from a supported region'):
+        Numbers(na_number='+44 20 7946 0958')
+
+
+def test_parse_error() -> None:
+    with pytest.raises(ValidationError, match='value is not a valid phone number'):
+        Numbers(phone_number='555 1212')
+
+
+def test_parsed_but_not_a_valid_number() -> None:
+    with pytest.raises(ValidationError, match='value is not a valid phone number'):
+        Numbers(phone_number='+1 555-1212')
+
+
+def test_json_schema() -> None:
+    assert Numbers.model_json_schema() == {
+        'title': 'Numbers',
+        'type': 'object',
+        'properties': {
+            'phone_number': {
+                'title': 'Phone Number',
+                'anyOf': [
+                    {'type': 'string', 'format': 'phone'},
+                    {'type': 'null'},
+                ],
+                'default': None,
+            },
+            'na_number': {
+                'title': 'Na Number',
+                'anyOf': [
+                    {'type': 'string', 'format': 'phone'},
+                    {'type': 'null'},
+                ],
+                'default': None,
+            },
+            'uk_number': {
+                'title': 'Uk Number',
+                'anyOf': [
+                    {'type': 'string', 'format': 'phone'},
+                    {'type': 'null'},
+                ],
+                'default': None,
+            },
+        },
+    }
+
+    class SomethingElse(BaseModel):
+        phone_number: Number
+
+    assert SomethingElse.model_json_schema() == {
+        'title': 'SomethingElse',
+        'type': 'object',
+        'properties': {
+            'phone_number': {
+                'title': 'Phone Number',
+                'type': 'string',
+                'format': 'phone',
+            }
+        },
+        'required': ['phone_number'],
+    }
```

Comment:
Amazing work @mZbZ but can you migrate this part of the tests to `tests/test_json_schema.py` in the tests directory so we can keep the tree of tests follow a pattern

Reply:
Thank you

No problem 

FYI, there is still a codecov report that is still running...

Reply:
> Thank you
> 
> No problem
> 
> FYI, there is still a codecov report that is still running...

somethings its get bugged and stick like this you can try another change or an empty commit and its will be fixed